File: server/hotrod/src/test/scala/org/infinispan/server/hotrod/transport/SingleByteFrameDecoderChannelInitializer.scala
package org.infinispan.server.hotrod.transport

import io.netty.channel.{Channel, ChannelInitializer}
import io.netty.handler.codec.FixedLengthFrameDecoder
import io.netty.handler.timeout.IdleStateHandler
import org.infinispan.server.core.ProtocolServer

/**
 * A channel pipeline factory for testing that will inject a fixed length frame encoder of 2 bytes before a
 * channel handler named <b>decoder</b>
 *
 * @author William Burns
 * @since 9.0
 */
trait SingleByteFrameDecoderChannelInitializer extends ChannelInitializer[Channel] {
   abstract override def initChannel(ch: Channel): Unit = {
      super.initChannel(ch)
      val pipeline = ch.pipeline
      pipeline.addBefore("decoder", "twoframe", new FixedLengthFrameDecoder(1))
   }
}

File: server/hotrod/src/test/scala/org/infinispan/server/hotrod/transport/SingleByteFrameDecoderChannelInitializer.java
package org.infinispan.server.hotrod.transport;

import org.infinispan.server.core.transport.NettyInitializer;

import io.netty.channel.Channel;
import io.netty.channel.ChannelPipeline;
import io.netty.handler.codec.FixedLengthFrameDecoder;

/**
 * A channel pipeline factory for testing that will inject a fixed length frame encoder of 2 bytes before a channel
 * handler named <b>decoder</b>
 *
 * @author William Burns
 * @since 9.0
 */
public class SingleByteFrameDecoderChannelInitializer implements NettyInitializer {
   @Override
   public void initializeChannel(Channel ch) throws Exception {
      ChannelPipeline pipeline = ch.pipeline();
      pipeline.addBefore("decoder", "twoframe", new FixedLengthFrameDecoder(1));
   }
}